composer/trainer/trainer.py

```diff
@@ -868,6 +868,12 @@ def __init__(
         self.state.evaluators = evaluators
 
         # Grad Clip Norm
+        if grad_clip_norm > 0:
+            warnings.warn(
+                DeprecationWarning(
+                    (f"Using the 'grad_clip_norm' field in Trainer is deprecated. PLease use"
+                     "the GradientClipping Algorithm in composer/algorithms/gradient_clipping")))
```

Comment:
```suggestion
                    (f"Using the 'grad_clip_norm' field in Trainer is deprecated. Please use"
                     "the GradientClipping Algorithm in composer.algorithms.gradient_clipping.")))
```

Reply:
Similiar to how we handled ssr, should we do something here where we add the gradient clipping algorithm if grad_clip_norm is not None?

Reply:
Then, would it be possible to rip out the following code from the trainer?

https://github.com/mosaicml/composer/blob/baeb171e796737aaea8dcb324113e7963a07f511/composer/trainer/trainer.py#L1717-L1721

That way, when we remove the deprecation, it'll just be in init and wherever we raise the warning -- would prefer not to have two implementations of grad clipping living around.

Reply:
ok done. check out [0e50ebb](https://github.com/mosaicml/composer/pull/1099/commits/0e50ebba2fe60a20cc09f4787bc98a472d0b7742)